### PR TITLE
Fix new use of async_main in balrogscript

### LINKED ISSUE #186: bug 1625053 - [balrogscript] post-async_main bustage
I landed https://github.com/mozilla-releng/scriptworker-scripts/pull/185 to fix, but we're still hitting

```
  File "/app/lib/python3.8/site-packages/balrogscript/script.py", line 187, in update_config
    config["api_root"] = config["server_config"][server]["api_root"]
TypeError: 'immutabledict' object does not support item assignment
```

Easy enough to fix: `config = dict(config)` before that line should do it. I think we should do a dev run before pushing to production again though, and potentially put in some `test_script` tests that don't mock quite as much.

### PR DESCRIPTION
**Pull request by Callek**
This fixes #186

### CODE REVIEW

File: balrogscript/src/balrogscript/script.py [hunk 1]
```diff
@@ -182,7 +183,7 @@ def set_readonly(task, config, auth0_secrets):
 
 # update_config {{{1
 def update_config(config, server="default"):
-    config = deepcopy(config)
+    config = deepcopy(dict(config))
```
Comment (escapewindow) [inline balrogscript/src/balrogscript/script.py]: I don't think we need the deepcopy if we `dict()`.

File: balrogscript/tests/conftest.py [hunk 2]
```diff
@@ -28,61 +29,73 @@ def release_manifest():
...
+            }
+        )
     finally:
         shutil.rmtree(tmpdir)
 
 
+@pytest.fixture(scope="function")
+def nightly_task(config):
+    with open(NIGHTLY_TASK_PATH, "r") as fh:
+        return json.load(fh)
```
Comment (escapewindow) [inline balrogscript/tests/conftest.py:81]: Nice. Are we able to use this fixture to fix this todo? https://github.com/mozilla-releng/scriptworker-scripts/blob/55b7bbfe5a026b8ef352859a2dae9e0663ba1c75/balrogscript/tests/test_script.py#L379-L380

**Review by escapewindow — APPROVED** [review]
Thanks! Have we tested in dev, or are you confident in this?

File: balrogscript/tests/conftest.py (hunk 2 above)
Comment (Callek) [reply balrogscript/tests/conftest.py:81]: Looks like yes!

**Pull request merged**